src/components/buoy-telemetry/BuoyMap.vue

```diff
@@ -81,53 +239,84 @@ onMounted(() => {
     scrollZoom: false,
   });
 
-  const geoJSON = {
-    type: "geojson",
-    data: {
-      id: "buoy",
-      type: "FeatureCollection",
-      features: store.coordinates.map(({ latitude, longitude }) => {
-        return {
-          type: "Feature",
-          geometry: {
-            type: "Point",
-            coordinates: [longitude, latitude],
-          },
-        };
-      }),
-    },
-  };
-
   map.value.on("load", function () {
     map.value.resize();
     map.value.addImage("buoy-marker", imageEl.value);
+    map.value.addImage("wind-speed-0", windSpeed0Image.value);
+    map.value.addImage("wind-speed-1", windSpeed1Image.value);
+    map.value.addImage("wind-speed-2", windSpeed2Image.value);
+    map.value.addImage("wind-speed-3", windSpeed3Image.value);
+    map.value.addImage("wind-speed-4", windSpeed4Image.value);
+    map.value.addImage("wind-speed-5", windSpeed5Image.value);
+    map.value.addImage("wind-speed-6", windSpeed6Image.value);
+    map.value.addImage("wind-speed-7", windSpeed7Image.value);
+    map.value.addImage("wind-speed-8", windSpeed8Image.value);
+    map.value.addImage("wind-speed-9", windSpeed9Image.value);
+    map.value.addImage("wind-speed-10", windSpeed10Image.value);
+
+    map.value.addSource("points", geoJSON.value);
 
-    map.value.addSource("points", geoJSON);
+    updateMap();
+  });
+});
 
+let markers = [];
+const updateMap = () => {
+  for (var b = 0; b < windSpeedBins.length - 1; b++) {
+    if (map.value.getLayer("svgfile" + b)) {
+      map.value.removeLayer("svgfile" + b);
+    }
+  }
+  if (map.value.getLayer("points")) {
+    map.value.removeLayer("points");
+  }
+  if (map.value.getSource("points")) {
+    map.value.removeSource("points");
+    map.value.addSource("points", geoJSON.value);
+  }
```

Comment:
I'd move the `addSource` outside of the if statement to parallel the structure of the rest of the block of code

```suggestion
  if (map.value.getSource("points")) {
    map.value.removeSource("points");
  }
map.value.addSource("points", geoJSON.value);
```

Reply:
good catch, thank you!